pkg/bucketclaim/bucketclaim.go

```diff
@@ -153,9 +160,12 @@ func (b *bucketClaimListener) provisionBucketClaimOperation(ctx context.Context,
 		}
 
 		bucketClass, err := b.bucketClasses().Get(ctx, bucketClassName, metav1.GetOptions{})
-		if err != nil {
+		if kubeerrors.IsNotFound(err) {
+			b.recordEvent(inputBucketClaim, v1.EventTypeWarning, events.FailedCreateBucket, err.Error())
+			return err
+		} else if err != nil {
 			klog.V(3).ErrorS(err, "Get Bucketclass Error", "name", bucketClassName)
-			return util.ErrInvalidBucketClass
+			return err
```

Comment:
Is there a reason that we don't want to record an event for all errors here? I think any failure is still a failure to create the bucket right?